src/kubernetes_operator/iam_mapping.py

```diff
@@ -0,0 +1,221 @@
+"""Kubernetes operator to manage IamIdentityMappings in the aws-config configmap."""
+import asyncio
+import logging
+from pathlib import Path
+
+import kopf
+import yaml
+from kubernetes import client, config
+from kubernetes.client.models.v1_config_map import V1ConfigMap
+
+logger = logging.getLogger("operator")
+
+try:
+    config.load_kube_config()
+except Exception as error:
+    logger.info("Could not load kubeconfig. Error is : %s.\nAssuming we are in a kubernetes cluster", error)
+    try:
+        config.load_incluster_config()
+    except Exception as error:
+        error_str = format(error)
+        raise Exception(f"No k8s config suitable, exiting ({error_str})") from error
+else:
+    logging.info("Using Kubernetes local configuration")
+
+API = client.CoreV1Api()
+custom_objects_API = client.CustomObjectsApi()
+GROUP = "iamauthenticator.k8s.aws"
+VERSION = "v1alpha1"
+PLURAL = "iamidentitymappings"
+
+
+@kopf.on.update(GROUP, VERSION, PLURAL)
+@kopf.on.create(GROUP, VERSION, PLURAL)
+async def create_mapping(spec: dict, diff: list, **_) -> None:
+    """Create/update an identity mapping in the aws-auth configmap with the corresponding IamIdentityMapping.
+
+    This method accepts mappings for userarn and rolearn with groups.
+
+    :param spec: The spec of the changed IamIdentityMapping
+    :param diff: The diff created by the changed identity
+    """
+
+    # Do nothing when we have no diff
+    if not len(diff) < 1:
```

Comment:
Hard to read condition IMHO, would probably be better to write it like this
```suggestion
    if not diff:
        return
```

Then de-indent the rest